README.md

```diff
@@ -8,19 +8,191 @@
+
+### Step 3: Create a HubSpot App
+
+   1. Now navigate to the Apps section from the left sidebar and click on the Create app button on the top right corner.
+
+![](https://github.com/SamudraUduwaka/module-ballerinax-hubspot.marketing.campaigns/blob/fb6603714f9b563b74775579577749ad62726576/docs/Resources/appSection.png)
+
+   2. Provide a public app name and description for your app.
+
+![](https://github.com/SamudraUduwaka/module-ballerinax-hubspot.marketing.campaigns/blob/fb6603714f9b563b74775579577749ad62726576/docs/Resources/namingApp.png)
+
+### Step 4: Setup Authentication
+
+   1. Move to the Auth tab.
+
+![](https://github.com/SamudraUduwaka/module-ballerinax-hubspot.marketing.campaigns/blob/fb6603714f9b563b74775579577749ad62726576/docs/Resources/auth.png)
+
+   2. In the Scopes section, add the following scopes for your app using the Add new scopes button.
+        - marketing.campaigns.read
+        - marketing.campaigns.revenue.read
+        - marketing.campaigns.write
+
+![](https://github.com/SamudraUduwaka/module-ballerinax-hubspot.marketing.campaigns/blob/fb6603714f9b563b74775579577749ad62726576/docs/Resources/marketingScopes.png)
+
+   3. In the Redirect URL section, add the redirect URL for your app. This is the URL where the user will be redirected after the authentication process. You can use localhost for testing purposes. Then hit the Create App button.
+
+![](https://github.com/SamudraUduwaka/module-ballerinax-hubspot.marketing.campaigns/blob/fb6603714f9b563b74775579577749ad62726576/docs/Resources/redirectURL.png)
+
+### Step 5: Get the Client ID and Client Secret
+
+Navigate to the Auth tab and you will see the Client ID and Client Secret for your app. Make sure to save these values.
+
+![](https://github.com/SamudraUduwaka/module-ballerinax-hubspot.marketing.campaigns/blob/fb6603714f9b563b74775579577749ad62726576/docs/Resources/clientId_secretId.png)
+
+### Step 6: Setup Authentication Flow
+
+Before proceeding with the Quickstart, ensure you have obtained the Access Token using the following steps:
+
+   1. Create an authorization URL using the following format:
+
+      ```
+      https://app.hubspot.com/oauth/authorize?client_id=<YOUR_CLIENT_ID>&scope=<YOUR_SCOPES>&redirect_uri=<YOUR_REDIRECT_URI>
+      ```
+   Replace the <YOUR_CLIENT_ID>, <YOUR_REDIRECT_URI> and <YOUR_SCOPES> with your specific value.
+
+   2. Paste it in the browser and select your developer test account to intall the app when prompted.
+
+![](https://github.com/SamudraUduwaka/module-ballerinax-hubspot.marketing.campaigns/blob/fb6603714f9b563b74775579577749ad62726576/docs/Resources/accountSelect.png)
+
+   3. A code will be displayed in the browser. Copy the code.
+
+      ```
+      Received code: na1-129d-860c-xxxx-xxxx-xxxxxxxxxxxx
+      ```
+
+   4. Run the following curl command. Replace the <YOUR_CLIENT_ID>, <YOUR_REDIRECT_URI> and <YOUR_CLIENT_SECRET> with your specific value. Use the code you received in the above step 3 as the CODE below
```

Comment:
Let's use the [Oxford comma](https://en.wikipedia.org/wiki/Serial_comma) in documentation.  

```suggestion
   4. Run the following curl command. Replace the `<YOUR_CLIENT_ID>`, `<YOUR_REDIRECT_URI>`, and `<YOUR_CLIENT_SECRET>` with your specific value. Use the code you received in the above step 3 as the `CODE` below
```